### ISSUE TITLE
Exclude some entries from production build

### ISSUE THREAD

**Issue by skyiea** [wix/yoshi#120]
**The case**:
I'm building (kind of) React components library, where components are exposed via CDN (it's BSM modules). All of exposed components are separate entry points.
In order to have proper dev experience I have (kind of) storybook entry, which is rendered in localhost and aggregates exposed components.

My yoshi config looks somewhat like this:
```
"entry": {
  "storybook": "./storybook.tsx",
  "landing-page": "./entries/LandingPage.entry.tsx",
  "visibility-widget": "./entries/VisibilityWidget.entry.tsx"
}
```

**The issue**:
The thing is that I don't want storybook entry _A)_ to be build in production environment (CI), _B)_ to be deployed to CDN. So *storybook* is *dev-only* entry (entry, which should be built solely in development environment).

**Solutions**:
There is a couple of solutions how to achieve / partially achieve the goal:
_Without **yoshi** modifications_:
1. Add `posttest` script, where manually remove unwanted files from `/dist` dir. It will fix B) case, though one'll need to know what exactly **yoshi** generates (`.js`, `.js.map`, `.min.js`, `.min.js.map`) and *how* it names bundles (`.bundle.js`). There is no option to configure the latter, so one would need to dig into **yoshi** source code as well.
2. Introduce additional index layer for dev-only entry, which will look like this:
```
if (process.ENV === 'development') {
   require('./storybook')
}
```
This solution won't fix neither cases, but at least storybook code (except webpack runtime and index layer IIFE code) won't go to CDN.

_What I imagine can be done inside **yoshi** to resolve this issue completely (fix A & B cases)_:
3. Add `devEntry` property, which will be an addition to and will behave the same way as `entry` property, but only if `yoshi build` is started with `process.ENV === 'development'`.
4. Allow defining *yoshi* configuration as separate JS file: `yoshi.config.js`. Inside the config it will be easy to manage entries however one'll want.

**Comment by shahata** [comment]
Let's say the first and last entry points are needed in dev mode only and the second is an actual entry point for production.

I'd go with having yoshi config like this:
```json
"entry": {
  "dev/storybook": "./storybook.tsx",
  "landing-page": "./entries/LandingPage.entry.tsx",
  "dev/visibility-widget": "./entries/VisibilityWidget.entry.tsx"
}
```

and then in posttest you can simply remove the whole dir: `"posttest": "rm -rf dist/statics/dev"`.
no need to know anything about how files are named then...

**Comment by shahata** [comment]
btw, I didn't understand why storybook is dev only. don't you want to publish it to a github page?

**Comment by skyiea** [comment]
@shahata Yes, with prefix thingy I won't need to know what files are generated. But still another issue I've described persists:
> I don't want storybook entry **to be build in production environment** (CI)

Regarding storybook - I can't see the reason to expose it. The components are not supposed to be used as presentational in other projects (this is not UI kit).

**Comment by yanivefraim** [comment]
@shahata - I see that more ppl need this feature  ( @eddierl , for example). Maybe we should add this to docs / have a built in solution inside Yoshi? WDYT?
